In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings('ignore')

In [3]:
df = pd.read_csv('C:\\Users\\DELL\\Desktop\Herovired All  Assignment\\Statstics Assignment\\property.csv')

In [4]:
df.head()

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000,S,Biggin,3/12/2016,2.5,3067,...,1,1.0,202,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019
1,Abbotsford,25 Bloomburg St,2,h,1035000,S,Biggin,4/2/2016,2.5,3067,...,1,0.0,156,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019
2,Abbotsford,5 Charles St,3,h,1465000,SP,Biggin,4/3/2017,2.5,3067,...,2,0.0,134,150.0,1900.0,Yarra,-37.8093,144.9944,Northern Metropolitan,4019
3,Abbotsford,40 Federation La,3,h,850000,PI,Biggin,4/3/2017,2.5,3067,...,2,1.0,94,NaN,NaN,Yarra,-37.7969,144.9969,Northern Metropolitan,4019
4,Abbotsford,55a Park St,4,h,1600000,VB,Nelson,4/6/2016,2.5,3067,...,1,2.0,120,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019


In [5]:
#column types
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  int64  
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  int64  
 10  Bedroom2       13580 non-null  int64  
 11  Bathroom       13580 non-null  int64  
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  int64  
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [6]:
df.isnull()

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,False,False,False,False,False,False,False,False,False,False,...,False,False,False,True,True,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,...,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,...,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,...,False,False,False,True,True,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,...,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13575,False,False,False,False,False,False,False,False,False,False,...,False,False,False,True,False,True,False,False,False,False
13576,False,False,False,False,False,False,False,False,False,False,...,False,False,False,False,False,True,False,False,False,False
13577,False,False,False,False,False,False,False,False,False,False,...,False,False,False,True,False,True,False,False,False,False
13578,False,False,False,False,False,False,False,False,False,False,...,False,False,False,False,False,True,False,False,False,False


In [7]:
# 2. Statistical summary (Mean, Min, Max, etc.
df.describe()

,Rooms,Price,Distance,Postcode,Bedroom2,Bathroom,Car,Landsize,BuildingArea,YearBuilt,Lattitude,Longtitude,Propertycount
count,13580.000000,1.358000e+04,13580.000000,13580.000000,13580.000000,13580.000000,13518.000000,13580.000000,7130.000000,8205.000000,13580.000000,13580.000000,13580.000000
mean,2.937997,1.075684e+06,10.137776,3105.301915,2.914728,1.534242,1.610075,558.416127,151.967650,1964.684217,-37.809203,144.995216,7454.417378
std,0.955748,6.393107e+05,5.868725,90.676964,0.965921,0.691712,0.962634,3990.669241,541.014538,37.273762,0.079260,0.103916,4378.581772
min,1.000000,8.500000e+04,0.000000,3000.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1196.000000,-38.182550,144.431810,249.000000
25%,2.000000,6.500000e+05,6.100000,3044.000000,2.000000,1.000000,1.000000,177.000000,93.000000,1940.000000,-37.856822,144.929600,4380.000000
50%,3.000000,9.030000e+05,9.200000,3084.000000,3.000000,1.000000,2.000000,440.000000,126.000000,1970.000000,-37.802355,145.000100,6555.000000
75%,3.000000,1.330000e+06,13.000000,3148.000000,3.000000,2.000000,2.000000,651.000000,174.000000,1999.000000,-37.756400,145.058305,10331.000000
max,10.000000,9.000000e+06,48.100000,3977.000000,20.000000,8.000000,10.000000,433014.000000,44515.000000,2018.000000,-37.408530,145.526350,21650.000000


In [8]:
# Check for missing values
display('Missing Value')
display(df.isnull().sum())

'Missing Value'

Suburb              0
Address             0
Rooms               0
Type                0
Price               0
Method              0
SellerG             0
Date                0
Distance            0
Postcode            0
Bedroom2            0
Bathroom            0
Car                62
Landsize            0
BuildingArea     6450
YearBuilt        5375
CouncilArea      1369
Lattitude           0
Longtitude          0
Regionname          0
Propertycount       0
dtype: int64

##### 📊 Missing Values Summary Report

The dataset is mostly clean and complete, with many important features having no missing values. Columns like Suburb, Address, Rooms, Price, Distance, Bathroom, and Location (Latitude/Longitude) are fully available, which is good for analysis and modeling.

However, a few columns contain missing data:

* BuildingArea → 6,450 missing values (very high)
* YearBuilt → 5,375 missing values (very high)
* CouncilArea → 1,369 missing values (moderate)
* Car → 62 missing values (low)

📌 Key Insight
Most features are reliable and ready to use.
But BuildingArea and YearBuilt have significant missing data, which may affect analysis accuracy.
These columns will require data cleaning or imputation before using them in models.
 
* ✅ Conclusion

The dataset is generally well-structured, but handling missing values in a few key columns is essential to ensure better data quality and accurate results.

In [9]:
# Drop high-risk column
df.drop('BuildingArea', axis=1, inplace=True)

# Fill numeric columns
df['YearBuilt'].fillna(df['YearBuilt'].median(), inplace=True)
df['Car'].fillna(df['Car'].median(), inplace=True)

# Fill categorical column
df['CouncilArea'].fillna(df['CouncilArea'].mode()[0], inplace=True)


# Fill numeric columns
df['Car'].fillna(df['Car'].median(), inplace=True)

### Missing Value Handling Strategy
🔴 1. BuildingArea (6450 missing – VERY HIGH)

Problem: Too many missing values

Best Options:

✅ If not very important → Drop the column

✅ If important → Use advanced imputation

* Why? Filling such a large gap with mean/median may introduce bias and wrong patterns.


2. YearBuilt (5375 missing – VERY HIGH)

Problem: Also heavily missing

Best Options:

✅ Convert to House Age instead of raw year
✅ Fill using median or group-based imputation


Insight: Age is more useful than raw year in prediction models.

3. CouncilArea (1369 missing – MODERATE)

Problem: Categorical column

Best Option:

Fill with mode (most frequent value)

Reason: Keeps category consistency




4. Car (62 missing – LOW)

Problem: Very small missing data

Best Options:

Fill with median (best for numeric)
OR drop those rows



In [10]:
# After Handle missing Values 
display(df.isnull().sum())

Suburb           0
Address          0
Rooms            0
Type             0
Price            0
Method           0
SellerG          0
Date             0
Distance         0
Postcode         0
Bedroom2         0
Bathroom         0
Car              0
Landsize         0
YearBuilt        0
CouncilArea      0
Lattitude        0
Longtitude       0
Regionname       0
Propertycount    0
dtype: int64

## Task 1. For the suburb of Altona, it is postulated that a typical property sells for $800,000.
• Use the data at hand to test this assumption.
• Is the typical property price really $800,000 or has it increased?
• Use a significance level of 5%.


In [ ]:

from scipy import stats

# filter for Altona
altona_prices = df[df['Suburb'] == 'Altona']['Price'].dropna()

# 2. Descriptive Statistics
mean_price = altona_prices.mean()
n_samples = len(altona_prices)
std_error = stats.sem(altona_prices)

print(f"--- Statistics for Altona ---")
print(f"Number of properties: {n_samples}")
print(f"Mean Price: ${mean_price:}")
print(f"Standard Error: {std_error:,.2f}")

--- Statistics for Altona ---
Number of properties: 74
Mean Price: $834830.4054054054
Standard Error: 33,891.54


In [12]:
# Perform One-Sample T-Test


# Hypothesis:
# Null Hypothesis (H0): Average Price = $800,000
# Alternative Hypothesis (H1): Average Price > $800,000 (Right-tailed test)
# Significance Level (alpha): 0.05

# Population mean under H0
pop_mean_h0 = 800000

# stats.ttest_1samp performs a two-tailed test by default. 
# We specify alternative='greater' for the right-tailed test (has it increased?)
t_stat, p_value = stats.ttest_1samp(altona_prices, pop_mean_h0, alternative='greater')

print(f"\n--- Hypothesis Test Results ---")
print(f"T-Statistic: {t_stat:.4f}")
print(f"P-Value: {p_value:.4f}")


alpha = 0.05
if p_value < alpha:
    print(f"\nConclusion: Since P-Value ({p_value:.4f}) < Alpha ({alpha}), we REJECT the null hypothesis.")
    print("There is sufficient evidence to suggest that the typical property price in Altona is significantly GREATER than $800,000.")
else:
    print(f"\nConclusion: Since P-Value ({p_value:.4f}) >= Alpha ({alpha}), we FAIL TO REJECT the null hypothesis.")
    print("There is not enough evidence to say that the typical property price in Altona has increased beyond $800,000.")



--- Hypothesis Test Results ---
T-Statistic: 1.0277
P-Value: 0.1537

Conclusion: Since P-Value (0.1537) >= Alpha (0.05), we FAIL TO REJECT the null hypothesis.
There is not enough evidence to say that the typical property price in Altona has increased beyond $800,000.


# Task 2. For the year 2016, is there any difference in the prices of properties sold in the
 summer months vs winter months?
• Consider months from October till March as winter months and rest as
summer months.
• Use a significance level of 5%

In [13]:
# Convert Date column
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)





In [14]:
# Null Hypothesis (H0): There is no difference in the mean prices of summer vs. winter months (mu{summer} = {winter}).
# Alternative Hypothesis (H1): there is a significant difference in the mean prices (mu{summer} \mu{winter}).
# Significance Level (alpha): 0.05 (5%).


# Winter: October, November, December, January, February, March.
# Summer: April, May, June, July, August, September.






# Filter for the year 2016
df_2016 = df[df['Date'].dt.year == 2016].copy()

# Categorize into Winter vs Summer based on user definition
# Winter: Oct(10), Nov(11), Dec(12), Jan(1), Feb(2), Mar(3)
winter_months = [10, 11, 12, 1, 2, 3]

def categorize_season(month):
    if month in winter_months:
        return 'Winter'
    else:
        return 'Summer'

df_2016['Season'] = df_2016['Date'].dt.month.apply(categorize_season)

#Extract prices for both groups
summer_prices = df_2016[df_2016['Season'] == 'Summer']['Price'].dropna()
winter_prices = df_2016[df_2016['Season'] == 'Winter']['Price'].dropna()

# Descriptive Statistics
print(f"--- 2016 Price Comparison ---")
print(f"Summer (Apr-Sep) Mean: ${summer_prices.mean():,.2f} (Count: {len(summer_prices)})")
print(f"Winter (Oct-Mar) Mean: ${winter_prices.mean():,.2f} (Count: {len(winter_prices)})")

# Perform Independent Two-Sample T-Test
# We use equal_var=False (Welch's T-test) as a safer default if variances are unequal
t_stat, p_value = stats.ttest_ind(summer_prices, winter_prices, equal_var=False)

print(f"\n--- Hypothesis Test Results ---")
print(f"T-Statistic: {t_stat:.4f}")
print(f"P-Value: {p_value:.4f}")


alpha = 0.05
if p_value < alpha:
    print(f"\nConclusion: Since P-Value ({p_value:.4f}) < Alpha ({alpha}), we REJECT the null hypothesis.")
    print("There IS a statistically significant difference in property prices between summer and winter months in 2016.")
else:
    print(f"\nConclusion: Since P-Value ({p_value:.4f}) >= Alpha ({alpha}), we FAIL TO REJECT the null hypothesis.")
    print("There is NO statistically significant difference in property prices between summer and winter months in 2016.")


--- 2016 Price Comparison ---
Summer (Apr-Sep) Mean: $1,048,054.73 (Count: 4036)
Winter (Oct-Mar) Mean: $1,116,647.59 (Count: 2300)

--- Hypothesis Test Results ---
T-Statistic: -3.9211
P-Value: 0.0001

Conclusion: Since P-Value (0.0001) < Alpha (0.05), we REJECT the null hypothesis.
There IS a statistically significant difference in property prices between summer and winter months in 2016.


## 3. For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have a car parking space?
• Use the column car in the dataset.
• Round off your answer to 3 decimal places

In [21]:
# Filter for Abbotsford
abbotsford_data = df[df['Suburb'] == 'Abbotsford']


# Calculate the probability 'p' of a property having NO car space (Car == 0)
# we drop NaNs in the Car column for an accurate proportion
total_properties = abbotsford_data['Car'].dropna().count()
no_car_properties = (abbotsford_data['Car'] == 0).sum()
p = no_car_properties / total_properties
print(f"--- Abbotsford Car Space Stats ---")
print(f"Total properties: {total_properties}")
print(f"Properties with 0 car spaces: {no_car_properties}")
print(f"Probability (p) of no car space: {p:.4f}")




#Binomial Distribution calculationy
# n = 10 properties, k = 3 properties with no car space
from scipy.stats import binom 
n = 10
k = 3
prob_exactly_3 = binom.pmf(k, n, p)
print(f"\n--- Binomial Probability ---")
print(f"Probability that exactly 3 out of 10 have no car space: {prob_exactly_3:.3f}")


--- Abbotsford Car Space Stats ---
Total properties: 56
Properties with 0 car spaces: 15
Probability (p) of no car space: 0.2679

--- Binomial Probability ---
Probability that exactly 3 out of 10 have no car space: 0.260


## Task 4. 
*In the suburb of Abbotsford, what are the chances of finding a property with 3   rooms? Round your answer to 3 decimal places.


In [15]:
# Filter 
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Count properties with exactly 3 rooms
three_room_count = (abbotsford_data['Rooms'] == 3).sum()

# Get the total count of properties in Abbotsford
total_abbotsford = len(abbotsford_data)

# Calculate the probability
probability = three_room_count / total_abbotsford

print(f"Total properties in Abbotsford: {total_abbotsford}")
print(f"Properties with 3 rooms: {three_room_count}")
print(f"Probability of finding a 3-room property: {probability:.3f}")


Total properties in Abbotsford: 56
Properties with 3 rooms: 20
Probability of finding a 3-room property: 0.357


# Task 5: In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places

In [16]:
# Count properties with exactly 2 bathrooms
two_bathroom_count = (abbotsford_data['Bathroom'] == 2).sum()

# Get the total count of properties in Abbotsford
total_abbotsford = len(abbotsford_data)

# Calculate the probability
probability = two_bathroom_count / total_abbotsford

print(f"Total properties in Abbotsford: {total_abbotsford}")
print(f"Properties with 2 bathrooms: {two_bathroom_count}")
print(f"Probability of finding a 2-bathroom property: {probability:.3f}")


Total properties in Abbotsford: 56
Properties with 2 bathrooms: 19
Probability of finding a 2-bathroom property: 0.339


# Task 6: One-Sample Hypothesis Test (Industry Pricing) A real estate firm claims that the average property price in Richmond is $1,000,000. Using the dataset, test whether the actual average price is significantly different from this claim at a 5% significance level. Clearly state:
* Null and alternative hypotheses
• Test statistic
• p-value
• Final business conclusion

In [17]:
# filter for Richmond
richmond_prices = df[df['Suburb'] == 'Richmond']['Price'].dropna()


# Null Hypothesis (H0): The average property price in Richmond is equal to $1,000,000 (mu = 1,000,000).
# Alternative Hypothesis (H1): The average property price in Richmond is significantly different from $1,000,000 (mu = 1,000,000).
# Significance Level (alpha): 0.05 (5%)



#Descriptive Statistics
mean_price = richmond_prices.mean()
n_samples = len(richmond_prices)

print(f"--- Statistics for Richmond ---")
print(f"Number of properties: {n_samples}")
print(f"Sample Mean Price: ${mean_price:,.2f}")

# Perform Two-Tailed One-Sample T-Test
# Population mean under H0
pop_mean_h0 = 1000000

# stats.ttest_1samp performs a two-tailed test by default
t_stat, p_value = stats.ttest_1samp(richmond_prices, pop_mean_h0)

print(f"\n--- Hypothesis Test Results ---")
print(f"Test Statistic (t): {t_stat:.4f}")
print(f"P-Value: {p_value:.4f}")

# Final Business Conclusion
alpha = 0.05
print("\n--- Final Business Conclusion ---")
if p_value < alpha:
    print(f"Since the p-value ({p_value:.4f}) is less than the significance level ({alpha}), we REJECT the null hypothesis.")
    print("Conclusion: There is sufficient evidence to conclude that the actual average property price in Richmond is significantly different from $1,000,000.")
else:
    print(f"Since the p-value ({p_value:.4f}) is greater than or equal to the significance level ({alpha}), we FAIL TO REJECT the null hypothesis.")
    print("Conclusion: There is not enough evidence to claim that the average property price in Richmond is significantly different from $1,000,000. The firm's claim is plausible.")


--- Statistics for Richmond ---
Number of properties: 260
Sample Mean Price: $1,083,564.42

--- Hypothesis Test Results ---
Test Statistic (t): 2.5795
P-Value: 0.0104

--- Final Business Conclusion ---
Since the p-value (0.0104) is less than the significance level (0.05), we REJECT the null hypothesis.
Conclusion: There is sufficient evidence to conclude that the actual average property price in Richmond is significantly different from $1,000,000.


## Task 7: Independent Two-Sample T-Test (Feature Impact)
Do properties with car parking sell at a higher average price than properties without
car parking, across the entire dataset? Use a 5% significance level and justify:
• Choice of test
• Interpretation of p-value
• Business implications for developers


In [18]:
# Null Hypothesis (H0): Properties with car parking have the same or lower average price compared to properties without car parking (mu{with_car} = mu{without_car}).
# Alternative Hypothesis (H1): Properties with car parking sell at a higher average price than properties without car parking (mu{with_car} > mu{without_car}).
# Significance Level (alpha): 0.05 (5%)




#drop rows where Price or Car is missing
clean_df = df.dropna(subset=['Price', 'Car'])

#Split into two groups
with_car = clean_df[clean_df['Car'] > 0]['Price']
without_car = clean_df[clean_df['Car'] == 0]['Price']

# Descriptive Statistics
print("--- Price Statistics ---")
print(f"With Car Parking: n = {len(with_car)}, Mean Price = ${with_car.mean():,.2f}")
print(f"Without Car Parking: n = {len(without_car)}, Mean Price = ${without_car.mean():,.2f}")

# Perform Independent Two-Sample T-Test
t_stat, p_value = stats.ttest_ind(with_car, without_car, equal_var=False, alternative='greater')

print("\n--- Hypothesis Test Results ---")
print(f"Test Statistic (t): {t_stat:.4f}")
print(f"P-Value: {p_value:}")

# Conclusion
alpha = 0.05
print("\n--- Conclusion ---")
if p_value < alpha:
    print(f"Since p-value < {alpha}, we REJECT the null hypothesis.")
    print("Properties with car parking sell at a statistically significant HIGHER average price.")
else:
    print(f"Since p-value >= {alpha}, we FAIL TO REJECT the null hypothesis.")
    print("There is no significant evidence to say properties with car parking sell at a higher price.")







--- Price Statistics ---
With Car Parking: n = 12554, Mean Price = $1,075,405.89
Without Car Parking: n = 1026, Mean Price = $1,079,088.01

--- Hypothesis Test Results ---
Test Statistic (t): -0.2159
P-Value: 0.585466113680856

--- Conclusion ---
Since p-value >= 0.05, we FAIL TO REJECT the null hypothesis.
There is no significant evidence to say properties with car parking sell at a higher price.


## Task: 8 Two-Way ANOVA (Location & Property Type)
Investigate whether property prices are influenced by:
• Suburb
• Type of property
• Interaction between suburb and property type
Use Two-Way ANOVA and explain which factors significantly affect price.


In [19]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

# clean missing values for relevant columns
clean_df = df.dropna(subset=['Price', 'Suburb', 'Type'])

# Build the Ordinary Least Squares (OLS) model
# The formula 'Price ~ C(Suburb) + C(Type) + C(Suburb):C(Type)' specifies the two main effects and their interaction.
# Note: This might take a few moments to run depending on the number of unique suburbs.
model = ols('Price ~ C(Suburb) + C(Type) + C(Suburb):C(Type)', data=clean_df).fit()

# Perform Two-Way ANOVA
anova_table = sm.stats.anova_lm(model, typ=2)

print("--- Two-Way ANOVA Results ---")
print(anova_table)


c:\Users\DELL\miniconda3\envs\ydata_profiling\lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 313, but rank is 150
  warnings.warn('covariance of constraints does not have full '
c:\Users\DELL\miniconda3\envs\ydata_profiling\lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 2, but rank is 0
  warnings.warn('covariance of constraints does not have full '


--- Two-Way ANOVA Results ---
                         sum_sq       df          F         PR(>F)
C(Suburb)          3.882913e+15    313.0  77.300361   0.000000e+00
C(Type)                     NaN      2.0        NaN            NaN
C(Suburb):C(Type)  6.470034e+14    626.0   6.440216  2.505093e-263
Residual           2.068639e+15  12890.0        NaN            NaN


c:\Users\DELL\miniconda3\envs\ydata_profiling\lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 626, but rank is 375
  warnings.warn('covariance of constraints does not have full '


### Task: 9. p-Value Interpretation (Decision Making)
A hypothesis test comparing prices across two suburbs results in a p-value of 0.032.
#### Answer:
* i. What does this p-value indicate?
* ii. Should the null hypothesis be rejected at α = 0.05?
* iii. How should a business stakeholder interpret this result?



This is a conceptual question, so you do not need to write any Python code for this. You can write the following answers directly into a Markdown cell in your Jupyter Notebook.

Here is how to answer the three parts:

**i. What does this p-value indicate?**
 
 A p-value of 0.032 indicates that there is only a 3.2% probability of observing a difference in average property prices as large as (or larger than) the one found in this sample by pure chance, assuming that the null hypothesis is true (i.e., assuming there is actually no real difference in prices between the two suburbs in the overall population). 

**ii. Should the null hypothesis be rejected at α = 0.05?**

**Yes**, the null hypothesis should be rejected. We compare the p-value (0.032) to the significance level (alpha = 0.05). Because 0.032 is less than 0.05, the result falls into the rejection region. We have sufficient statistical evidence to reject the null hypothesis.

**iii. How should a business stakeholder interpret this result?**

A business stakeholder (like a real estate investor or developer) should interpret this to mean that the price difference between these two suburbs is **statistically significant and real**, not just a random fluke in the data. 


*   **Actionable Insight:** They can confidently use this information to make strategic decisions. For example, if one suburb is proven to be significantly cheaper, it might represent a better area for entry-level buyers or a targeted marketing campaign. Conversely, the more expensive suburb might be prioritized for luxury developments.

## 10. Industry-Style Hypothesis Validation (Policy Decision)
A housing policy group believes that properties with more than 2 bathrooms
command a premium price.
Design and execute a statistical test to validate this claim:
• Identify the correct test
• State hypotheses
• Report p-value
• Give a clear recommendation to policymakers


In [20]:


#drop missing values for Price and Bathroom
clean_df = df.dropna(subset=['Price', 'Bathroom'])

#Split into the two groups
more_than_two_baths = clean_df[clean_df['Bathroom'] > 2]['Price']
two_or_less_baths = clean_df[clean_df['Bathroom'] <= 2]['Price']

#Show the basic means
print("--- Average Prices ---")
print(f"More than 2 Bathrooms: ${more_than_two_baths.mean():,.2f}")
print(f"2 or Fewer Bathrooms: ${two_or_less_baths.mean():,.2f}")

#Perform the Independent Right-Tailed T-Test
t_stat, p_value = stats.ttest_ind(more_than_two_baths, two_or_less_baths, equal_var=False, alternative='greater')

print("\n--- Statistical Test Results ---")
print(f"Test Statistic (t): {t_stat:.4f}")
print(f"P-Value: {p_value:.4e}")  # Uses scientific notation as the p-value will be very small

#Conclusion
if p_value < 0.05:
    print("\nResult: REJECT the Null Hypothesis. Properties with >2 bathrooms command a premium.")
else:
    print("\nResult: FAIL TO REJECT the Null Hypothesis.")


--- Average Prices ---
More than 2 Bathrooms: $1,882,824.20
2 or Fewer Bathrooms: $1,007,347.94

--- Statistical Test Results ---
Test Statistic (t): 28.8002
P-Value: 3.6902e-137

Result: REJECT the Null Hypothesis. Properties with >2 bathrooms command a premium.
